File: src/main/java/ast20201/project/service/CreditService.java
package ast20201.project.service;

import java.math.BigDecimal;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import ast20201.project.repository.UserRepository;

@Service
public class CreditService {

	@Autowired
	UserRepository userRepository;

	public BigDecimal getCredits(long userId) {
        return userRepository.getUser(userId).getCredits();
    }
}

File: src/main/webapp/components/credit/credit.service.js
(function () {
    'use strict';

    angular
        .module('app')
        .factory('creditService', creditService);

        creditService.$inject = ['$http'];

    function creditService($http) {
        var service = {
            getCredits: getCredits
        };

        return service;

        function getCredits() {
            return $http.get(`/api/credits`);
        }
    }
})();